libsplinter/src/events/ws.rs

```diff
@@ -447,11 +471,22 @@ fn do_shutdown(
 pub struct Context<T: ParseBytes<T> + 'static> {
     igniter: Igniter,
     ws: WebSocketClient<T>,
+    reconnect_count: u64,
+    last_reconnect: SystemTime,
+    wait: Duration,
+    reactor_running: Arc<AtomicBool>,
```

Comment:
I don't think `reactor_running` is necessary since `igniter` has the same variable. If `Context` needs to know if it's reactor is running it should call a method from `igniter`.

Additionally, I wonder if it would make more sense for the igniter to check if its reactor is still running an emit an error when context or a callback call `start_ws` or `send`. I don't think that needs to be addressed in this PR, just curious what people think.

Reply:
yeah, I can use the reactor_running in the igniter, and will make that change. But I need the Context to know that the reactor is still running, otherwise, if the gameroom daemon is stopped while the context is waiting before attempt to reconnect to a ws server, the shutdown would be delayed until the wait time is over
https://github.com/Cargill/splinter/blob/cc31433dfb446221104452e0d0437a7e9779fdc1/libsplinter/src/events/ws.rs#L541-L554

Reply:
updated